src/app/components/projects/import-project-dialog/import-project-dialog.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, OnInit, Inject, ViewChild } from '@angular/core';
+import { MatStepper, MatDialogRef, MAT_DIALOG_DATA } from "@angular/material";
+import { FileUploader, ParsedResponseHeaders, FileItem } from 'ng2-file-upload';
+import { Server } from '../../../models/server';
+import { v4 as uuid } from 'uuid';
+
+@Component({
+    selector: 'app-import-project-dialog',
+    templateUrl: 'import-project-dialog.component.html',
+    styleUrls: ['import-project-dialog.component.css'],
+})
+export class ImportProjectDialogComponent implements OnInit {
+    uploader: FileUploader;
+    server : Server;
+    projectName : string;
+    isImportEnabled : boolean = false;
+    isFinishEnabled : boolean = false;
+    errorMessage : string;
+
+    @ViewChild('stepper') stepper: MatStepper;
+  
+    constructor(
+      public dialogRef: MatDialogRef<ImportProjectDialogComponent>,
+      @Inject(MAT_DIALOG_DATA) public data: any){}
+  
+    ngOnInit(){
+      this.uploader = new FileUploader({});
+      this.uploader.onAfterAddingFile = (file) => { file.withCredentials = false; };
+    }
+  
+    uploadProjectFile(event) : void{
+      this.projectName = event.target.files[0].name.split(".")[0];
+      this.isImportEnabled = true;
+    }
+  
+    onImportClick() : void{
+      if(this.validateProjectName()){
```

Comment:
We may consider more generic approach of form validation, like:
http://jasonwatmore.com/post/2018/05/10/angular-6-reactive-forms-validation-example
http://jasonwatmore.com/post/2018/05/11/angular-6-template-driven-forms-validation-example